File: Unidad_2/05_Lambda_ejercicios/01_Lambda_ejercicio_c1/Principal.java
public class Principal{
    public static void main(String[] args) {
        Principal.engine((int x, int y)->(long x1, long y2) -> x1 + y2);
    }
    // Sobrecarga de métodos
    private static void engine(CalculadoraInt cal){
        int x = 2, y = 4;
        CalculadoraLong resultado = cal.calcular(x,y);
        System.out.println("Resultado = " + resultado.calcular((long)x, (long)y));
    }
}

File: 2 _oportunidad/_05_ejercicios/lambda_ambiguedad_solucion_2/Principal.py
import math

class Principal:
    @staticmethod
    def main(args):
        #Casting a nivel de Interfaz
        Principal._engine(lambda x, y : x+y)
        Principal._engine(lambda x, y : x*y)

        Principal._engine(lambda x, y : x/y)
        Principal.engine(lambda x, y : x-y)
        Principal._engine(lambda x, y : math.fmod(x, y))
    @staticmethod
    def _engine(cal):
        x =2
        y =4
        resultado = cal.calcular(x,y)
        print(resultado)
    @staticmethod
    def _engine(cal):
        x =2
        y =4
        resultado = cal.calcular(x,y)
        print(resultado)
